lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -439,3 +451,108 @@ getPoolPerformanceEstimate sp dl rp pid = do
+-- removed.
+--
+-- TODO: For reasons of performance, this summary does not
+-- fill in the 'performanceEstimate' field of each stake pool.
+-- As it is not needed for the new reward computation,
+-- it will be removed in the future.
+getStakePoolsSummary'
+    :: BF.MonadBlockfrost m
+    => Tracer m Log
+    -> m StakePoolsSummary
+getStakePoolsSummary' tr = do
+    let mkPoolMap pids = Map.fromList [ (fromPoolId p,p) | p <- pids ]
+    pools <- mkPoolMap <$> getAllPages BF.listPools'
+
+    rp <- getRewardParams
+    let getPool pid = do
+            info <- fromPoolInfo rp <$> BF.getPool pid
+            traceWith tr $ MsgRewardInfoPool (fromPoolId pid) info
+            pure info
+
+    StakePoolsSummary rp <$> mapM getPool pools
+
+getRewardParams :: BF.MonadBlockfrost m => m RewardParams
+getRewardParams = do
+    BF.ProtocolParams{..} <- BF.getLatestEpochProtocolParams
+    BF.Network{..} <- BF.getNetworkInfo
+    BF.EpochInfo{..} <- BF.getLatestEpoch
+    let BF.NetworkSupply{..} = _networkSupply
+        eta
+            | _protocolParamsDecentralisationParam >= 0.8 = 1
+            | otherwise = 1 -- wrong value
+                -- NOTE: The correct value is  blocksMade % expectedBlocks
+                -- But since we are long after d = 1, we ignore that here.
+        deltaR1 = rationalToCoinViaFloor $
+            min 1 eta
+                * realToFrac _protocolParamsRho
+                * fromIntegral _supplyReserves
+        rationalToCoinViaFloor :: Rational -> Coin
+        rationalToCoinViaFloor = Coin . floor
+        rPot = fromLovelaces _epochInfoFees <> deltaR1
+        rp = RewardParams
+            { nOpt =
+                fromIntegral _protocolParamsNOpt
+            , a0 =
+                realToFrac _protocolParamsA0
+            , r =
+                rPot
+            , totalStake =
+                fromLovelaces _supplyTotal
+            }
+    pure rp
+
+fromPoolInfo :: RewardParams -> BF.PoolInfo -> RewardInfoPool
+fromPoolInfo rp BF.PoolInfo{..} = RewardInfoPool
+    { stakeRelative = clipToPercentage $
+        fromIntegral _poolInfoActiveStake / _totalStake
+    , ownerPledge =
+        fromLovelaces _poolInfoDeclaredPledge
+    , ownerStake =
+        fromLovelaces _poolInfoLivePledge
+    , ownerStakeRelative = clipToPercentage $
+        fromIntegral _poolInfoLivePledge / _totalStake
+    , cost =
+        fromLovelaces _poolInfoFixedCost 
+    , margin = clipToPercentage $
+        realToFrac _poolInfoMarginCost
+    , performanceEstimate =
+        1.0 -- assume perfect performance here
+    }
+  where
+    _totalStake = fromIntegral . unCoin $ totalStake rp
+
+{-------------------------------------------------------------------------------
+    Type conversions
+-------------------------------------------------------------------------------}
+fromLovelaces :: BF.Lovelaces -> Coin
+fromLovelaces = Coin . toEnum . fromIntegral
+
+-- BF.PoolId is Bech32 encoded
+fromPoolId :: BF.PoolId -> PoolId
```

Comment:
I'd expect this conversion to be implemented as a `FromBlocfrost` typeclass instance as it includes error type in result.

Reply:
I'd like to do that, but as `fromPoolId` is used inside `getStakePoolsSummary'`, I would need some way to embed `BlockfrostError` into the `BlockfrostClientT` monad — but I can only embed `BF.BlockfrostError`. 🤔 I could have `getStakePoolsSummary'` return an `Either`. What would you advise?


… There is a reason why I'm not too fond of tracking errors in the type level — at least not without sufficient combinator support. 😅

Reply:
Agreed on the following:
- don't expose conversion error in type but make sure that if it fails then error message is clear (e.g. contains context description)
- solution applied here should be also considered for the similar cases in this module.